internal/pod/imagepuller.go

```diff
@@ -55,33 +55,33 @@ func NewImagePuller(client client.Client, scheme *runtime.Scheme) ImagePuller {
 	}
 }
 
-func (ipi *imagePullerImpl) CreatePullPod(ctx context.Context, imageSpec *kmmv1beta1.ModuleImageSpec,
-	micObj *kmmv1beta1.ModuleImagesConfig) error {
+func (ipi *imagePullerImpl) CreatePullPod(ctx context.Context, name, namespace, imageToPull string, oneTimePod bool,
+	imageRepoSecret *v1.LocalObjectReference, owner metav1.Object) error {
 
-	pullPodTypeLabeValue := pullPodUntilSuccess
-	if imageSpec.Build != nil || imageSpec.Sign != nil {
-		pullPodTypeLabeValue = pullPodTypeOneTime
+	pullPodTypeLabelValue := pullPodUntilSuccess
+	if oneTimePod {
+		pullPodTypeLabelValue = pullPodTypeOneTime
 	}
 
 	imagePullSecrets := []v1.LocalObjectReference{}
-	if micObj.Spec.ImageRepoSecret != nil {
-		imagePullSecrets = []v1.LocalObjectReference{*micObj.Spec.ImageRepoSecret}
+	if imageRepoSecret != nil {
+		imagePullSecrets = []v1.LocalObjectReference{*imageRepoSecret}
 	}
 
 	pullPod := v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: micObj.Name + "-pull-pod-",
-			Namespace:    micObj.Namespace,
+			GenerateName: name + "-pull-pod-",
+			Namespace:    namespace,
 			Labels: map[string]string{
-				moduleImageLabelKey: micObj.Name,
-				pullPodTypeLabelKey: pullPodTypeLabeValue,
+				moduleImageLabelKey: name,
```

Comment:
`moduleImageLabelKey` should be changed to something more generic as it will now always be a MIC.